src/test/java/org/junitpioneer/jupiter/params/SimpleAggregatorTests.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.params;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junitpioneer.testkit.PioneerTestKit.executeTestMethodWithParameterTypes;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Month;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.assertion.PioneerAssert;
+
+class SimpleAggregatorTests {
+
+	@ParameterizedTest
+	@CsvSource({ "23, M, Bob", "24, F, Jane" })
+	void testAggregator(@Aggregate Person person) {
+		if (person.gender == Gender.M) {
+			assertThat(person.age).isEqualTo(23);
+			assertThat(person.name).isEqualTo("Bob");
+		} else {
+			assertThat(person.age).isEqualTo(24);
+			assertThat(person.name).isEqualTo("Jane");
+		}
+	}
+
+	@ParameterizedTest
+	@ValueSource(ints = { 13, 17, 19 })
+	void testBoxing(@Aggregate Boxed boxed) {
+		assertThat(boxed.value).isLessThan(20);
+	}
+
+	@ParameterizedTest
+	@CsvSource({ "John, 2023-07-16", "Bob, 1959-02-21", "Jane, 1977-05-03" })
+	void testLocalDate(@Aggregate Human human) {
+		assertThat(human.name).isNotBlank();
+		assertThat(human.birthday).isAfter(LocalDate.of(1900, 1, 1));
```

Comment:
I stumbled a bit over the time assertion. Can we make these a bit more precise, e.g., using `isIn(String... localDateAsString)`?